src/lib.rs

```diff
@@ -601,3 +597,149 @@ impl<'lib> IRCompiler<'lib> {
+            };
+
+            vec![push_constants, indirect_identifier]
+        };
+
+        let static_samplers = [
+            create_static_sampler(
+                IRFilter::IRFilterMinMagMipPoint,
+                IRTextureAddressMode::IRTextureAddressModeWrap,
+                0,
+                None,
+            ),
+            create_static_sampler(
+                IRFilter::IRFilterMinMagMipPoint,
+                IRTextureAddressMode::IRTextureAddressModeClamp,
+                1,
+                None,
+            ),
+            create_static_sampler(
+                IRFilter::IRFilterMinMagMipLinear,
+                IRTextureAddressMode::IRTextureAddressModeWrap,
+                2,
+                None,
+            ),
+            create_static_sampler(
+                IRFilter::IRFilterMinMagMipLinear,
+                IRTextureAddressMode::IRTextureAddressModeClamp,
+                3,
+                None,
+            ),
+            create_static_sampler(
+                IRFilter::IRFilterMinMagMipLinear,
+                IRTextureAddressMode::IRTextureAddressModeBorder,
+                4,
+                None,
+            ),
+            create_static_sampler(
+                IRFilter::IRFilterAnisotropic,
+                IRTextureAddressMode::IRTextureAddressModeWrap,
+                5,
+                Some(2),
+            ),
+            create_static_sampler(
+                IRFilter::IRFilterAnisotropic,
+                IRTextureAddressMode::IRTextureAddressModeWrap,
+                6,
+                Some(4),
+            ),
+        ];
+
+        let desc_1_1 = IRRootSignatureDescriptor1 {
+            flags: IRRootSignatureFlags::IRRootSignatureFlagCBVSRVUAVHeapDirectlyIndexed,
+            num_parameters: parameters.len() as u32,
+            p_parameters: parameters.as_ptr(),
+            num_static_samplers: static_samplers.len() as u32,
+            p_static_samplers: static_samplers.as_ptr(),
+        };
+
+        let desc = IRVersionedRootSignatureDescriptor {
+            version: IRRootSignatureVersion::IRRootSignatureVersion_1_1,
+            u: IRVersionedRootSignatureDescriptor_u { desc_1_1 },
+        };
+
+        let root_sig = IRRootSignature::create_from_descriptor(&lib, &desc)?;
+
+        // Cross-compile to Metal
+        let mut mtl_binary = IRMetalLibBinary::new(&lib)?;
+        let obj = IRObject::create_from_dxil(&lib, &dxil_binary)?;
+        let mut c = IRCompiler::new(&lib)?;
+        c.set_global_root_signature(&root_sig);
+        let mtllib = c.alloc_compile_and_link(&[b"main\0"], &obj)?;
+        mtllib.get_metal_lib_binary(IRShaderStage::IRShaderStageCompute, &mut mtl_binary);
+
+        Ok(mtl_binary.get_byte_code())
+    }
+}
+
+pub fn compile_dxil_to_metallib_from_path(
+    path_compiler_lib: &Path,
+    path_dxil: &Path,
+) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
+    // Load DXIL binary from path
+    let dxil_binary = std::fs::read(path_dxil)?;
```

Comment:
Since you're only reading a file into binary, let the user pass a byte slice. This would make it compatible with our asset pipeline that does not give you access to plain files.

Perhaps the same for the compiler library itself. Even better: how expensive is it to load the compiler and interface for every shader? Perhaps we need to provide an object-based API of this sort:

```rust
let compiler = MetalCompiler::new("path/to/libirmetallibbinary.so")?; // TODO: Should there be a function with a sensible default path? Is it available on the system?

let _compiled = compiler.compile(&[0x13, 0x37, ...])?;
```